src/mpid/ch4/generic/am/mpidig_am_part_utils.h

```diff
@@ -9,20 +9,35 @@
 #include "ch4_impl.h"
 
 /* Request status used for partitioned comm.
- * Updated as atomic cntr because AM callback and local operation may be
+ *
+ * For sender:
+ *     1. Psend_init:    status = 0, send active message MPIDIG_PART_SEND_INIT
+ *     2. Start (local): status += 1
+ *     3. receive CTS:   status += 1
+ *     4. send data when status == 2 and (all) partitions ready
+ *     5. send complete: status -= 2
+ *     ready for next start.
+ *
+ *     Note that 2 and 3 can take place in any order. Next round step 3 may happen between 4 and 5.
+ *
+ * For receirver:
+ *     1. Precv_init:    status = 0
+ *     2. receive MPIDIG_PART_SEND_INIT: status +=1
+ *     3. match and send CTS
+ *     4. Start (local): status += 1
+ *     5. receive all data and complete: status -= 1
+ *     ready for next start.
+ *
+ *     Note that 2 and 4 can take place in any order, and 3 takes place after 2 or 4, whichever is later.
+ *     After first round, it no longer need matching, so no longer need step 1, 2, and 3.
+ */
+
+/* Updated as atomic cntr because AM callback and local operation may be
  * concurrently performed by threads. */
-#define MPIDIG_PART_REQ_UNSET 0 /* Initial status set at psend|precv_init */
-#define MPIDIG_PART_REQ_PARTIAL_ACTIVATED 1     /* Intermediate status (unused).
-                                                 * On receiver means either (1) request matched at initial round
-                                                 * or (2) locally started;
-                                                 * On sender means either (1) receiver started
-                                                 * or (2) locally started. */
-#define MPIDIG_PART_REQ_CTS 2   /* Indicating receiver is ready to receive data
-                                 * (requests matched and start called) */
-
-/* Status to deactivate at request completion */
-#define MPIDIG_PART_SREQ_INACTIVE 0     /* Sender need 2 increments to be CTS: start and remote start */
-#define MPIDIG_PART_RREQ_INACTIVE 1     /* Receiver need 1 increments to be CTS: start */
+
+#define MPIDIG_PART_STATUS_UNSET 0      /* Initial status set at psend|precv_init */
+#define MPIDIG_PART_STATUS_CAN_SEND 2   /* sender side started and got CTS, can send data */
+#define MPIDIG_PART_STATUS_CAN_CTS  2   /* receiver side started and matched, can send CTS */
 
```

Comment:
Can we embed sender-side and receiver-side hint into the second and the third statuses respectively? I.e., `MPIDIG_PART_STATUS_CAN_SEND` is used only on the sender side. In addition, `MPIDIG_PART_STATUS_UNSET` is unused after the change, but I'd recommend we use it instead of magic number (see above comment).

Additional macros might be added to express the -`1` and -`2` numbers when completing a request.